app/src/lib/shells/linux.ts

```diff
@@ -114,6 +127,14 @@ export function launch(
     case Shell.Tilix:
     case Shell.Gnome:
       return spawn(foundShell.path, ['--working-directory', path])
+    case Shell.PowerShellCore:
+      const psCoreCommand = `"Set-Location -LiteralPath '${path}'"`
+      return spawn(foundShell.path, [
+        'pwsh',
```

Comment:
I currently see this error when trying to open a repository in PowerShell Core on Ubuntu 16.04:

> debug: [ui] [PowerShell Core] stderr: 'The argument 'pwsh' is not recognized as the name of a script file. Check the spelling of the name, or if a path was included, verify that the path is correct and try again.'
> debug: [ui] [PowerShell Core] exit code: 64

I thought removing the standalone `pwsh` here would be the fix (it's how PowerShell should work on Windows), and the error does go away, but then I don't see a shell being displayed and I don't see any other errors, so I'm not sure what's happened here.

Reply:
Removing `pwsh` should work. (Sorry for including it!)

Does `pwsh -NoExit -Command "Set-Location -LiteralPath '(pathname)'"` on bash shell work there? It seems to take some time at first time.

Reply:
> Does `pwsh -NoExit -Command "Set-Location -LiteralPath '(pathname)'"` on bash shell work there? 

Haven't tried it out, but the fact that you're running it within an existing shell means it's not quite the same setup as what Desktop is trying to do.

I think we need to leverage `$TERM` here to run `pwsh` within the context of a terminal emulator to get it appearing (hat tip to @daviwil for the clue):

<img width="469" src="https://user-images.githubusercontent.com/359239/38220537-ffe44614-371d-11e8-845c-25008d0327fc.png">

Reply:
EDIT: actually, `$TERM` isn't set when launching Desktop like you would a regular application (this only really worked because I was doing `yarn start:dev` to start Desktop). Back to the drawing board.

Reply:
Actually, if `pwsh` on Linux depends on running in an existing terminal, then it doesn't really make sense to have this as a menu option for Linux.

Any objections to just deferring first-class Linux support until someone figures out what that should look like @saschanaz @daviwil?

Reply:
Turns out that `require("child_process").spawnSync("/usr/bin/pwsh", ['-NoExit'])` just emits stdout to pipe and then exits.